osh/builtin_process2.py

```diff
@@ -215,8 +215,11 @@ def Run(self, cmd_val):
         # NOTE: This happens if we have '8' or '9' in the input too.
         stderr_line("osh warning: umask with symbolic input isn't implemented")
         return 1
-      else:
-        posix.umask(new_mask)
-        return 0
+      except TypeError:
+        # Should be unreachable
+        raise AssertionError()
+
+      posix.umask(new_mask)
```

Comment:
Ah OK I think this is because mycpp ignores "else" in try/except/else!

Because C++ doesn't have that construct

We could make that an error to find these bugs

Reply:
Great point, thanks for finding that!  I filed a bug about it

https://github.com/oilshell/oil/issues/1362

mycpp would be a good thing to talk about, I kind of want to rewrite it as Pea, with Python 3.10 pattern matching (which you mentioned! ):   https://oilshell.zulipchat.com/#narrow/stream/121539-oil-dev/topic/First.20step.20for.20Pea.3F

It's a bit messy, but I have been able to incrementally add all the features I want, like field masks last week.

(Jesse and I did a lot of working cleaning up and fixing the runtime in the last 3 months, but we haven't cleaned up the translator.  It's kind of burning a hole in my brain :) Seems fun but not sure I can justify it )